### PR TITLE
Add EnumTags class and generate tags for all enums 

### PR DESCRIPTION
**Pull request by fmatthew5876**
Introduces an enum_tags<E,N> class which provides enum to string functionality. Generator creates these for all RPG enums.

### CODE REVIEW

File: src/enum_tags.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
+/*
```
Comment (carstene1ns) [inline src/enum_tags.h]: Please add this file to [Makefile.am](https://github.com/EasyRPG/liblcf/blob/master/Makefile.am#L115). This is especially needed for distribution archives.

**Comment by fmatthew5876** [comment]
Added to Makefile.am

**Comment by Ghabry** [comment]
Sorry for this basic question but how am I supposed to use this? The template looks pretty straight-forward but I'm lost.

As an example lets say I want a tag-iterator RPG::Item::Type. (accessing kTypeTags directly doesn't work, gives linker error)

**Comment by fmatthew5876** [comment]
> Sorry for this basic question but how am I supposed to use this? The template looks pretty straight-forward but I'm lost.
> 
> As an example lets say I want a tag-iterator for RPG::Item::Type.

You can use it like this:

```c++
for (auto* x: RPG::Skill::kTypeTags) {
    std::cout << x << std::endl;
}
```

>  (accessing kTypeTags directly doesn't work, gives linker error)

My fault. I forgot to provide definitions for these. static variables require a definition, although a lot of times you can get away without one in release builds.

**Pull request merged**